javaagent/src/main/java/io/opentelemetry/javaagent/OpenTelemetryAgent.java

```diff
@@ -67,6 +67,17 @@ public static void agentmain(String agentArgs, Instrumentation inst) {
       System.err.println("ERROR " + thisClass.getName());
       ex.printStackTrace();
     }
+
+    try {
+      // Call oshi.SystemInfo.getCurrentPlatformEnum() to activate SystemMetrics.
+      // Oshi instrumentation will intercept this call and enable SystemMetrics.
+      Class<?> oshiSystemInfoClass =
+          ClassLoader.getSystemClassLoader().loadClass("oshi.SystemInfo");
+      Method getCurrentPlatformEnumMethod = oshiSystemInfoClass.getMethod("getCurrentPlatformEnum");
+      getCurrentPlatformEnumMethod.invoke(null);
+    } catch (Throwable ex) {
+      // OK
+    }
```

Comment:
I'm ok with this approach for now, and we can log an issue to improve this in the future (ideally the agent wouldn't know about the instrumentation), @iNikem @anuraaga?

Reply:
@trask Do we have any gotchas for using system classloader here? It'd be good to understand well if there are cases we don't support right now.

Following up with the classloader callback approach sounds fine though.

Reply:
Added #1565 to track this